x-pack/legacy/plugins/siem/server/lib/detection_engine/signals/single_bulk_create.ts

```diff
@@ -139,8 +143,9 @@ export const singleBulkCreate = async ({
     delete errorCountsByStatus['409']; // Duplicate signals are expected
 
     if (!isEmpty(errorCountsByStatus)) {
```

Comment:
Right now you're outputting a lot of `409 conflict messages` which are not what we are interested in as 409 conflicts are ok because we do a de-duplication.

What you want to do is report per each line of your log file is any non-409 messages such as the 400 errors you are getting.

I would remove this and instead add these lines of code right before the `if (!isEmpty(errorCountsByStatus)) {` like so:

```ts
    itemsWithErrors
      .filter(item => item.create.status !== 409) // remove any 409 conflict error codes as those are expected due to de-duplication
      .forEach(item => {
        if (item.create.error != null) {
          logger.error(
            `Rule id: "${id}" rule_id: "${ruleParams.ruleId}" name: "${name}" has an error response with the reason of: ${item.create.error.reason}`
          );
        } else {
          // return the full item if there is no reason but typically Elastic Search should always return a reason
          logger.error(
            `Rule id: "${id}" rule_id: "${
              ruleParams.ruleId
            }" name: "${name}" has an error response with the reason of: ${JSON.stringify(item)}`
          );
        }
      });

if (!isEmpty(errorCountsByStatus)) {
```

I don't see anything wrong with logging out all of the errors you are seeing as we are interested in seeing them all and fixing them.

Reply:
it makes sense to filter out 409. I just pushed a commit to include that. Thanks.

There is no need to print out the individual errors. Let's just think the scenarios we add a noisy rule that generates 1000s of bad requests and this would lead to a lot loggings that can cause the server running out of local disk space quickly. 
I think it is ok to fix a few errors in each iteration. We do not need to fix all the errors at once, and also in the most common cases, the error messages are the same so no need to print them in a loop and repeat the same for 1000s time, right?

Reply:
I agree and talked with team members so we are going to aggregate on the message reason and output that instead of the status code.

That should keep logs from filling up but also give us more robust error reporting for logs when responses are coming back unexpected with errors such as networking errors, gateway errors, or other misc things.

The PR is this one if you want to watch for it:
https://github.com/elastic/kibana/pull/63513

Reply:
looks good to me. thanks